packages/@aws-cdk/aws-s3-deployment/test/test.bucket-deployment.ts

```diff
@@ -118,6 +119,54 @@ export = {
     test.done();
   },
 
+  'distribution can be used to provide a CloudFront distribution for invalidation'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const bucket = new s3.Bucket(stack, 'Dest');
+    const distribution = new cloudfront.CloudFrontWebDistribution(stack, 'Distribution', {
+      originConfigs: [
+        {
+          s3OriginSource: {
+            s3BucketSource: bucket
+          },
+          behaviors : [ {isDefaultBehavior: true}]
+        }
+      ]
+    });
+
+    // WHEN
+    new s3deploy.BucketDeployment(stack, 'Deploy', {
+      source: s3deploy.Source.asset(path.join(__dirname, 'my-website.zip')),
+      destinationBucket: bucket,
+      distribution,
+      distributionPaths: ['/images/*']
```

Comment:
test what happens if "distribution" is provided but "distributionPaths" is omitted